tests/rgen/Microsoft.Macios.Generator.Tests/Classes/Data/ExpectedConstructorTests.cs

```diff
@@ -0,0 +1,176 @@
+	///     } else {
+	///         Handle = ObjCRuntime.Messaging.IntPtr_objc_msgSendSuper_CGRect (this.SuperHandle, initWithFrame, frame);
+	///     }
+	/// }
+	/// ]]></code>
+	///     </example>
+	/// </remarks>
+	[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+	[EditorBrowsable (EditorBrowsableState.Advanced)]
+	protected ConstructorTests (global::Foundation.NSObjectFlag t) : base (t) {}
+
+	/// <summary>A constructor used when creating managed representations of unmanaged objects. Called by the runtime.</summary>
+	/// <param name="handle">Pointer (handle) to the unmanaged object.</param>
+	/// <remarks>
+	///     <para>
+	///         This constructor is invoked by the runtime infrastructure (<see cref="ObjCRuntime.Runtime.GetNSObject(System.IntPtr)" />) to create a new managed representation for a pointer to an unmanaged Objective-C object.
+	///         Developers should not invoke this method directly, instead they should call <see cref="ObjCRuntime.Runtime.GetNSObject(System.IntPtr)" /> as it will prevent two instances of a managed object pointing to the same native object.
+	///     </para>
+	/// </remarks>
+	[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+	[EditorBrowsable (EditorBrowsableState.Advanced)]
+	protected internal ConstructorTests (global::ObjCRuntime.NativeHandle handle) : base (handle) {}
+
+	[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+	[DesignatedInitializer]
+	public ConstructorTests (string path, bool isDir) : base (global::Foundation.NSObjectFlag.Empty)
+	{
+		if (path is null)
+			global::ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (path));
+		var nspath = global::CoreFoundation.CFString.CreateNative (path);
+		if (IsDirectBinding) {
+			InitializeHandle (global::ObjCRuntime.Messaging.NativeHandle_objc_msgSend_NativeHandle_bool (this.Handle, global::ObjCRuntime.Selector.GetHandle ("initFileURLWithPath:isDirectory:"), nspath, isDir ? (byte) 1 : (byte) 0), "initFileURLWithPath:isDirectory:");
+		} else {
+			InitializeHandle (global::ObjCRuntime.Messaging.NativeHandle_objc_msgSendSuper_NativeHandle_bool (this.SuperHandle, global::ObjCRuntime.Selector.GetHandle ("initFileURLWithPath:isDirectory:"), nspath, isDir ? (byte) 1 : (byte) 0), "initFileURLWithPath:isDirectory:");
+		}
+		global::CoreFoundation.CFString.ReleaseNative (nspath);
+	}
+
+	[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
+	public ConstructorTests (string scheme, string host, string path) : base (global::Foundation.NSObjectFlag.Empty)
+	{
+		if (scheme is null)
+			global::ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (scheme));
+		var nsscheme = global::CoreFoundation.CFString.CreateNative (scheme);
+		if (host is null)
+			global::ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (host));
```

Comment:
This will leak `nsscheme` is `host` is null. All the validation should be done first, before we start creating native strings.

Reply:
I'll have to fix that in methods too, will do a diff PR with that fix, will then do the creation. Will post the PR in a follow up comment.